src/answers-headless.ts

```diff
@@ -270,10 +270,19 @@ export default class AnswersHeadless {
     return results;
   }
 
-  executeFilterSearch(sectioned: boolean, fields: SearchParameterField[]): Promise<FilterSearchResponse> {
+  async executeFilterSearch(
+    query: string,
+    sectioned: boolean,
+    fields: SearchParameterField[]
+  ): Promise<FilterSearchResponse | undefined> {
+    const verticalKey = this.state.vertical.key;
+    if (!verticalKey) {
+      console.error('no verticalKey supplied for filter search');
+      return;
```

Comment:
Does this `return` here count as `Promise<undefined>`?

Reply:
yep